src/zabapgit_repo_impl.prog.abap

```diff
@@ -295,6 +307,20 @@ CLASS lcl_repo_online IMPLEMENTATION.
 
   ENDMETHOD.  " rebuild_local_checksums.
 
+  METHOD delete_initial_online_repo.
+
+    IF me->is_offline( ) = abap_false AND me->get_sha1_local( ) IS INITIAL.
+
+      lcl_app=>repo_srv( )->delete( me ).
+
+      IF iv_commit = abap_true.
+        COMMIT WORK.
+      ENDIF.
```

Comment:
but if there is an error the exception will be catched in lcl_gui->handle_action with an rollback
and LCL_SERVICES_REPO->CLONE isn`t the only caller